include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -697,7 +738,72 @@ class turn_in_piece_visitor
         }
         return 1; // inside
     }
-#endif
+
+public :
+
+    template <typename Turn, typename Piece>
+    static inline int apply(Turn const& turn, Piece const& piece)
+    {
+        // We don't know (yet)
+        if (piece.is_convex)
+        {
+            return in_convex_piece(turn, piece);
+        }
+        else
+        {
+
+            // TODO: this point_in_geometry is a performance-bottleneck here and
+            // will be replaced completely by extending analyse_piece functionality
+            return detail::within::point_in_geometry(turn.robust_point, piece.robust_ring);
```

Comment:
You should pass CS-specific winding strategy extracted from buffer strategy into this algorithm.

Reply:
Thank you

Reply:
Also here: done, can you verify if this is what you intended?